clarity_ext/service/artifact_service.py

```diff
@@ -88,7 +88,6 @@ def all_output_files(self):
         files = (outp for outp in outputs
                  if outp.output_type == Artifact.OUTPUT_TYPE_RESULT_FILE)
         ret = list(utils.unique(files, lambda f: f.id))
-        assert len(ret) == 0 or isinstance(ret[0], ResultFile)
```

Comment:
Yes, I think that needs to be addressed too, but decided to limit the effect of this commit.

This assert was failing because of an earlier commit, which needs to be cleaned up. I did only remove the assert that was causing tests to fail, and I added a ticket on the rest (LIMS-769). I think it would make sense to investigate that when working on the ticket.